### PR TITLE
[draft] Pulsar modeling module and hook

### PR DESCRIPTION
**Pull request by celiotine**

### CODE REVIEW

File: posydon/binary_evol/SN/step_SN.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-"""Supernova step.
+f"""Supernova step.
```
Comment (devimisra) [inline posydon/binary_evol/SN/step_SN.py]: typo "f"?

Comment (celiotine) [reply posydon/binary_evol/SN/step_SN.py]: Thanks for catching!

File: posydon/binary_evol/pulsar_modeling.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        I = self.moment_inertia
+        R = self.radius
+        B_i = self.Bfield
+
+        ## evolve the NS B-field
+        B_f = (B_i - B_min) * np.exp(-delta_t/tau_d) + B_min
+        self.Bfield = B_f
+
+        ## evolve the NS spin
+        A = 8*np.pi*R**6*np.sin(alpha)**2/(3*mu_0*const.clight**3*I)
```
Comment (devimisra) [inline posydon/binary_evol/pulsar_modeling.py:192]: Shouldn't there be a negative sign here for A in line 126? Unless you took care of it in the equation for final spin.

File: posydon/binary_evol/pulsar_modeling.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        G = const.standard_cgrav     ## gravitational constant [cm^3 g^-1 s^-2]
+        delta_Md = 0.025*const.Msun  ## magnetic field mass decay scale [g]
+        B_min = 1e8                  ## minimum Bfield strength at which Bfield decay ceases [G]
+        mu_0 = 1                        ## permeability of free space [has value unity in cgs]
+
+        M_i = self.mass              ## mass of the NS before accretion [g]
+        R = self.radius              ## radius of the NS
+        B_i = self.Bfield            ## B-field of the NS before accretion [G]
+
+        R_alfven = (2*np.pi**2/(G*mu_0**2))**(1/7) * (R**6/Mdot*M_i**(1/2))**(2/7) * B_i**(4/7) ## Alfven radius
```
Comment (devimisra) [inline posydon/binary_evol/pulsar_modeling.py]: Why use the NS Eddington mass rate to calculate the Alfven radius, and not the NS mass accretion rate?

**Comment by mkruckow** [comment]
We will wait with this PR until v3.

**Pull request closed without merging**